mantidproject/mantid
# Deploy Workbench on Windows

**Description of work.**

- Necessary Workbench DLLs for startup have been added to Mantid's deployment
- launch_workbench startup Python script has been added, with supporting Batch and PowerShell wrapper scripts to use the correct Python from Mantid's Installation
- The PowerShell startup script is compiled into an EXE during CMake Generation, this is done as a step in WindowsNSIS if ENABLE_WORKBENCH and PACKAGE_WORKBENCH are set to true, and PowerShell is available. The executable is compiled for PowerShell 2.0 and it should be possible to
run it on every Windows 7 and up.
- Workbench statup scripts installation has been moved to WindowsNSIS
- Added placeholder icons for Workbench, used only in the Windows installation

**To test:**
1. Do ALL_BUILD of Mantid in `Release` configuration. With `ENABLE_WORKBENCH=true`, `PACKAGE_WORKBENCH=true`, and `ENABLE_CPACK=true`.
  
   > The DEBUG configuration will not run after being installed as the `debug` DLLs are not packaged in the installation.
   - Check if:
    - [ ] PowerShell was found and has successfully generated `launch_workbench.exe.install`

2. After that is complete run CPack to create the install package
   - PowerShell one-liner that avoids Chocolatey's CPACK. Assumes CMake in default install directory. Must be run from the `build` directory (where the `CPackConfig.cmake` file is)
   - `& 'C:\Program Files\CMake\bin\cpack.exe' -C Release --config .\CPackConfig.cmake`
   - If getting `Chocolatey v0.10.0 File specified is either not found or not a .nuspec file. '--config .\CPackConfig.cmake'` then remove `C:\ProgramData\chocolatey\bin` from your `PATH` environment variable. Oneliner: ` $env:PATH = $env:PATH.Replace("C:\ProgramData\chocolatey\bin;", "");`

3. Install the created package, things to look for:
   - [ ] Workbench shortcut on Desktop and in Start Menu
   - [ ] Running the shortcut actually runs Mantid Workbench
   - [ ] Running the other shortcuts actually runs what they should run
   - [ ] Go into Mantid's install directory and run `launch_workbench.exe`, this should also run the Workbench

4. Generating the project without PowerShell, but with Workbench enabled:
   - Run the CMake Generation without `powershell.exe` in path. Make sure to DELETE the cache, otherwise PowerShell.exe will still be found (CMake does not seem to rerun `find_program` on programs that are already found)
   - PS one-liner, assumes default CMake installation directory, and starts the GUI without `powershell` in path: 
   - `$env:PATH = $env:PATH.Replace("C:\windows\System32\WindowsPowerShell\v1.0\;", ""); & "C:\Program Files\CMake\bin\cmake-gui.exe"`

5. Install the created no-workbench-shortcut package, things to look for:
   - [ ] No Workbench shortcut anywhere
   - [ ] Running the other shortcuts still runs what they should run

Fixes #23263 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
